tests/test_ensembles/test_voting_ensemble.py

```diff
@@ -86,33 +89,79 @@ def test_forecast_interface(example_tsds: TSDataset, catboost_pipeline: Pipeline
+    ensemble.fit(ts=example_tsds)
+    start_idx = 20
+    end_idx = 30
+    prediction = ensemble.predict(
+        start_timestamp=example_tsds.index[start_idx], end_timestamp=example_tsds.index[end_idx]
+    )
+    assert isinstance(prediction, TSDataset)
+    assert len(prediction.df) == end_idx - start_idx + 1
+
+
+def test_vote_default_weights(simple_df: TSDataset, naive_pipeline_1: Pipeline, naive_pipeline_2: Pipeline):
+    """Check that VotingEnsemble gets average during vote."""
     ensemble = VotingEnsemble(pipelines=[naive_pipeline_1, naive_pipeline_2])
     ensemble.fit(ts=simple_df)
-    forecast = ensemble.forecast()
+    forecasts = Parallel(n_jobs=ensemble.n_jobs, backend="multiprocessing", verbose=11)(
+        delayed(ensemble._forecast_pipeline)(pipeline=pipeline) for pipeline in ensemble.pipelines
+    )
+    forecast = ensemble._vote(forecasts=forecasts)
     np.testing.assert_array_equal(forecast[:, "A", "target"].values, [47.5, 48, 47.5, 48, 47.5, 48, 47.5])
     np.testing.assert_array_equal(forecast[:, "B", "target"].values, [11, 12, 11, 12, 11, 12, 11])
 
 
-def test_forecast_values_custom_weights(simple_df: TSDataset, naive_pipeline_1: Pipeline, naive_pipeline_2: Pipeline):
-    """Check that VotingEnsemble gets average."""
+def test_vote_custom_weights(simple_df: TSDataset, naive_pipeline_1: Pipeline, naive_pipeline_2: Pipeline):
+    """Check that VotingEnsemble gets average during vote."""
     ensemble = VotingEnsemble(pipelines=[naive_pipeline_1, naive_pipeline_2], weights=[1, 3])
     ensemble.fit(ts=simple_df)
-    forecast = ensemble.forecast()
+    forecasts = Parallel(n_jobs=ensemble.n_jobs, backend="multiprocessing", verbose=11)(
+        delayed(ensemble._forecast_pipeline)(pipeline=pipeline) for pipeline in ensemble.pipelines
+    )
+    forecast = ensemble._vote(forecasts=forecasts)
     np.testing.assert_array_equal(forecast[:, "A", "target"].values, [47.25, 48, 47.25, 48, 47.25, 48, 47.25])
     np.testing.assert_array_equal(forecast[:, "B", "target"].values, [10.5, 12, 10.5, 12, 10.5, 12, 10.5])
 
 
-def test_forecast_prediction_interval_interface(example_tsds, naive_pipeline_1, naive_pipeline_2):
-    """Test the forecast interface with prediction intervals."""
-    ensemble = VotingEnsemble(pipelines=[naive_pipeline_1, naive_pipeline_2], weights=[1, 3])
-    ensemble.fit(example_tsds)
-    forecast = ensemble.forecast(prediction_interval=True, quantiles=[0.025, 0.975])
-    for segment in forecast.segments:
-        segment_slice = forecast[:, segment, :][segment]
-        assert {"target_0.025", "target_0.975", "target"}.issubset(segment_slice.columns)
-        assert (segment_slice["target_0.975"] - segment_slice["target_0.025"] >= 0).all()
+def test_forecast_calls_vote(example_tsds: TSDataset, naive_pipeline_1: Pipeline, naive_pipeline_2: Pipeline):
+    ensemble = VotingEnsemble(pipelines=[naive_pipeline_1, naive_pipeline_2])
+    ensemble.fit(ts=example_tsds)
+    ensemble._vote = MagicMock()
```

Comment:
Do we actually need MagicMock here, why not just Mock?

Reply:
As i understand MagicMock is useful testing the magic methods, but here we don't need it